packages/atlas/src/views/studio/VideoWorkspace/VideoWorkspace.tsx

```diff
@@ -1,46 +1,36 @@
 import React, { useCallback, useEffect, useState } from 'react'
 import { CSSTransition } from 'react-transition-group'
+import useMeasure from 'react-use-measure'
 
 import { DrawerHeader } from '@/components/DrawerHeader'
+import { SvgControlsCancel } from '@/components/_icons'
 import { useDisplayDataLostWarning } from '@/hooks/useDisplayDataLostWarning'
 import { useHeadTags } from '@/hooks/useHeadTags'
-import {
-  VideoWorkspaceAssetsCache,
-  VideoWorkspaceState,
-  useVideoWorkspace,
-  useVideoWorkspaceData,
-} from '@/providers/videoWorkspace'
+import { useMediaMatch } from '@/hooks/useMediaMatch'
+import { VideoWorkspaceFormStatus, useVideoWorkspace, useVideoWorkspaceData } from '@/providers/videoWorkspace'
 import { cVar } from '@/styles'
-import { computeFileHash } from '@/utils/hashing'
 
-import { Container, DrawerOverlay } from './VideoWorkspace.style'
-import { VideoWorkspaceForm } from './VideoWorkspaceForm'
+import { VideoForm } from './VideoForm'
+import { Container, DrawerOverlay, ScrollContainer, StyledActionBar } from './VideoWorkspace.style'
+import { useHandleVideoWorkspaceSubmit } from './hooks'
 
 export const VideoWorkspace: React.FC = React.memo(() => {
-  // videoWorkspace state
-  const { videoWorkspaceState, setVideoWorkspaceState, editedVideoInfo } = useVideoWorkspace()
+  const [formStatus, setFormStatus] = useState<VideoWorkspaceFormStatus | null>(null)
+  const [actionBarHeight, setActionBarHeight] = useState(0)
+
+  const { isWorkspaceOpen, setIsWorkspaceOpen, editedVideoInfo } = useVideoWorkspace()
+  const { tabData } = useVideoWorkspaceData()
+
   const { openWarningDialog } = useDisplayDataLostWarning()
-  const { tabData } = useVideoWorkspaceData(editedVideoInfo)
-  const [assetsCache, setAssetsCache] = useState<VideoWorkspaceAssetsCache>(null)
+
+  const handleSubmit = useHandleVideoWorkspaceSubmit()
 
   const isEdit = !editedVideoInfo?.isDraft
   const headTags = useHeadTags(isEdit ? 'Edit video' : 'New video')
-  const anyVideoTabCachedAssets = !!(
-    assetsCache &&
-    (assetsCache.thumbnail.cropContentId || assetsCache.video.contentId)
-  )
-
-  // transaction management
-  const [thumbnailHashPromise, setThumbnailHashPromise] = useState<Promise<string> | null>(null)
-  const [videoHashPromise, setVideoHashPromise] = useState<Promise<string> | null>(null)
-  const [dialogState, setDialogState] = useState<VideoWorkspaceState>('unset')
-
-  useEffect(() => {
```

Comment:
This was the fix for opening form directly from `/video-workspace`. It doesn't open now because on initial render `isWorkspaceOpen === true` and `CSSTransition` doesn't see change.

Reply:
I think it's something else, when you open `/video-workspace`, the route changes to `/videos`, that wouldn't be caused by lack of this effect I think

Reply:
Okay that route change happens on prod as well :P